### DISCUSSION TITLE
`CropForegroundd` is only cropping in one dimension

### DISCUSSION THREAD

**Discussion by ahxmeds** [Project-MONAI/MONAI#6351] [Q&A] [answered]
Hi all, 

I have been playing around with `CropForegroundd` transform and found that for 2D PET/CT slices, it was only cropping the foreground in one of the dimensions. For instance, I use the following transforms: 

```
transforms_nocrop = Compose(
    [
        LoadImaged(['CT', 'PT'], image_only=False),
        ScaleIntensityRanged('CT', a_min=-1024, a_max=1024, b_min=0, b_max=1, clip=True),
        ScaleIntensityd('PT', minv=0, maxv=1)
    ]
)

transforms_crop = Compose(
    [
        LoadImaged(['CT', 'PT'], image_only=False),
        ScaleIntensityRanged('CT', a_min=-1024, a_max=1024, b_min=0, b_max=1, clip=True),
        ScaleIntensityd('PT', minv=0, maxv=1),
        CropForegroundd(keys=['CT', 'PT'], source_key='CT'),
    ]
)

tr1 =  CacheDataset(train_sample, transform=transforms_nocrop, num_workers=12)
tr2 =  CacheDataset(train_sample, transform=transforms_crop, num_workers=12)

ct1, pt1 = tr1[0]['CT'], tr1[0]['PT']
ct2, pt2 = tr2[0]['CT'], tr2[0]['PT']

ct1_new = np.copy(ct1)
ct1_new[ct1_new>0] = 1
pt1_new = np.copy(pt1)
pt1_new[pt1_new>0] = 1

ct2_new = np.copy(ct2)
ct2_new[ct2_new>0] = 1
pt2_new = np.copy(pt2)
pt2_new[pt2_new>0] = 1

fig, ax = plt.subplots(2,4, figsize=(10,7))
ax[0][0].imshow(ct1)
ax[0][1].imshow(pt1)
ax[0][2].imshow(ct1_new)
ax[0][3].imshow(pt1_new)
ax[1][0].imshow(ct2)
ax[1][1].imshow(pt2)
ax[1][2].imshow(ct2_new)
ax[1][3].imshow(pt2_new)

ax[0][0].set_title(f'CT:\n{ct1.shape}')
ax[0][1].set_title(f'PT:\n{pt1.shape}')
ax[0][2].set_title('CT mask')
ax[0][3].set_title('PT mask')

ax[1][0].set_title(f'CT-FGcrop:\n{ct2.shape}')
ax[1][1].set_title(f'PT-FGcrop:\n{pt2.shape}')
ax[1][2].set_title('CT-FGcrop mask')
ax[1][3].set_title('PT-FGcrop mask')

plt.show()
plt.close('all')
   
```

The plot is shown below. The top row shows the non-cropped images (via `transforms_nocrop` ), while the bottom row shows the images after foreground cropping (via `transforms_crop`). The cropping is with respect to the `CT` slice. In the 3rd and 4th columns, I am also plotting the masks for CT and PET slices showing regions for which the values on the slices are non-zero. As you can see in the bottom row, the cropping is asymmetrical (see the image under `CT-FGcrop mask`). Please let me know what the issue could be.

![image](https://user-images.githubusercontent.com/48228716/231599381-7fde2121-01fc-4035-be33-124e2048e708.png)

**Comment by KumoLiu** [comment, marked as answer]
Hi @ahxmeds, seems you didn't add a channel dim. Could you please try to add `EnsureChannelFirst` after `LoadImaged`?
Hope it can help you, thanks!
